HTSlib 1.21 loaded from /nfs/sw/easybuild/software/htslib/1.21
Samtools 1.21 loaded from /nfs/sw/easybuild/software/samtools/1.21

The following have been reloaded with a version change:
  1) GCCcore/12.3.0 => GCCcore/13.3.0
  2) zlib/1.2.13-GCCcore-12.3.0 => zlib/1.3.1-GCCcore-13.3.0

WARNING:pairtools:Headerless input, please, add the header by `pairtools header generate` or `pairtools header transfer`
Traceback (most recent call last):
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/bin/pairtools", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/__init__.py", line 183, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/sort.py", line 128, in sort
    sort_py(
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/sort.py", line 175, in sort_py
    header = headerops.append_new_pg(header, ID=UTIL_NAME, PN=UTIL_NAME)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/lib/headerops.py", line 333, in append_new_pg
    raise Exception("Input file is not valid .pairs, has no header or is empty.")
Exception: Input file is not valid .pairs, has no header or is empty.
Traceback (most recent call last):
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/bin/pairtools", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/__init__.py", line 183, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/sort.py", line 128, in sort
    sort_py(
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/sort.py", line 161, in sort_py
    instream = fileio.auto_open(
               ^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/lib/fileio.py", line 153, in auto_open
    return open(path, mode)
           ^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/var/spool/slurmd/job5700393/tmp'
WARNING:pairtools:Headerless input, please, add the header by `pairtools header generate` or `pairtools header transfer`
WARNING:pairtools:Pairs file appears not to be sorted, dedup might produce wrong results.
Traceback (most recent call last):
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/bin/pairtools", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/__init__.py", line 183, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/dedup.py", line 326, in dedup
    dedup_py(
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/dedup.py", line 488, in dedup_py
    header = headerops.append_new_pg(header, ID=UTIL_NAME, PN=UTIL_NAME)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/lib/headerops.py", line 333, in append_new_pg
    raise Exception("Input file is not valid .pairs, has no header or is empty.")
Exception: Input file is not valid .pairs, has no header or is empty.
WARNING:pairtools:Headerless input, please, add the header by `pairtools header generate` or `pairtools header transfer`
WARNING:pairtools:Pairs file appears not to be sorted, dedup might produce wrong results.
Traceback (most recent call last):
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/bin/pairtools", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/__init__.py", line 183, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/dedup.py", line 326, in dedup
    dedup_py(
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/dedup.py", line 488, in dedup_py
    header = headerops.append_new_pg(header, ID=UTIL_NAME, PN=UTIL_NAME)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/lib/headerops.py", line 333, in append_new_pg
    raise Exception("Input file is not valid .pairs, has no header or is empty.")
Exception: Input file is not valid .pairs, has no header or is empty.
WARNING:pairtools:Headerless input, please, add the header by `pairtools header generate` or `pairtools header transfer`
Traceback (most recent call last):
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/bin/pairtools", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/__init__.py", line 183, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/split.py", line 44, in split
    split_py(pairsam_path, output_pairs, output_sam, **kwargs)
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/cli/split.py", line 80, in split_py
    header = headerops.append_new_pg(header, ID=UTIL_NAME, PN=UTIL_NAME)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nfs/sw/easybuild/software/Python/3.12.3-GCCcore-13.3.0/lib/python3.12/site-packages/pairtools/lib/headerops.py", line 333, in append_new_pg
    raise Exception("Input file is not valid .pairs, has no header or is empty.")
Exception: Input file is not valid .pairs, has no header or is empty.
[main_samview] fail to read the header from "-".
[W::hts_set_opt] Cannot change block size for this format
samtools sort: failed to read header from "-"
[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 1589404 sequences (240000004 bp)...
[M::process] read 1589404 sequences (240000004 bp)...
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17627, 221341, 17302, 17857)
[M::mem_pestat] analyzing insert size distribution for orientation FF...
[M::mem_pestat] (25, 50, 75) percentile: (988, 2576, 5033)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13123)
[M::mem_pestat] mean and std.dev: (3265.50, 2680.27)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 17168)
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat] (25, 50, 75) percentile: (153, 209, 340)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 714)
[M::mem_pestat] mean and std.dev: (236.91, 125.79)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 901)
[M::mem_pestat] analyzing insert size distribution for orientation RF...
[M::mem_pestat] (25, 50, 75) percentile: (871, 2425, 4923)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13027)
[M::mem_pestat] mean and std.dev: (3174.65, 2726.49)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 17079)
[M::mem_pestat] analyzing insert size distribution for orientation RR...
[M::mem_pestat] (25, 50, 75) percentile: (1013, 2541, 5074)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 13196)
[M::mem_pestat] mean and std.dev: (3279.88, 2680.06)
[M::mem_pestat] low and high boundaries for proper pairs: (1, 17257)
[M::mem_process_seqs] Processed 1589404 reads in 636.400 CPU sec, 27.164 real sec
[E::hts_open_format] Failed to open file "/gpfs/commons/home/sliu/ALS_MicroC_analyses/output/03_mapped_pairs_per_sample/Sample_GWF2162_mapped_pairs/mapped.PT.bam" : No such file or directory
samtools index: failed to open "/gpfs/commons/home/sliu/ALS_MicroC_analyses/output/03_mapped_pairs_per_sample/Sample_GWF2162_mapped_pairs/mapped.PT.bam": No such file or directory
